sound/soc/sof/intel/hda.c

```diff
@@ -613,6 +613,11 @@ static int hda_init_caps(struct snd_sof_dev *sdev)
 	if (!(interface_mask & BIT(SOF_DAI_INTEL_ALH)))
 		goto skip_soundwire;
 
+	/* Skip SoundWire in nocodec mode */
+	if (IS_ENABLED(CONFIG_SND_SOC_SOF_NOCODEC_DEBUG_SUPPORT) &&
+	    sof_debug_check_flag(SOF_DBG_FORCE_NOCODEC))
+		goto skip_soundwire;
+
```

Comment:
"SoundWire is not needed" but does it help to actually remove the scan?

I don't recall what "nocodec" means but we certainly have cases where we don't have actual SoundWire codecs connected. Is this the case that nocodec really means SSP/DMIC only?

Reply:
Yes, nocodec is SSP+DMIC. We need HDA for host DMA (and for Link with ACE2+).
What @lyakh faced is that a device which fails to probe with SDW cannot be turned to nocodec due to the fact that SDW fails. This should not prevent nocodec as it is irrelevant.

I don't know the details, but I think it is due to misconfiguration of ACPI table for SDW versus reality.

Yes, this happens in development environment, so things are moving here and there and the nocodec is a developer configuration.